### ISSUE TITLE
gsutil rsync "invalid literal"

### ISSUE THREAD

**Issue by ellisio** [GoogleCloudPlatform/gsutil#209]
Hello,

This has been working just fine for the last several days. However, today, this is no longer working.

```
gsutil -m rsync -d -r gs://liquidcompass/installer-files files
```

This is a directory within our main bucket that contains multiple sub-directories of files for our server provisioning process.

When running this command I get the following:

```
Building synchronization state...
Skipping cloud sub-directory placeholder object gs://liquidcompass/installer-files/
Starting synchronization
invalid literal for int() with base 10: ''
CommandException: 1 files/objects could not be copied/removed.
```

Cheers,
-Andrew

**Comment by fabio-rigato** [comment]
hello,
I'm facing the same issue.

**gsutil version: 4.61**

`Exception in thread Thread-12:B]                                                
Traceback (most recent call last):
  File "/snap/google-cloud-sdk/179/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/snap/google-cloud-sdk/179/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/daisy_chain_wrapper.py", line 200, in PerformDownload
    decryption_tuple=self.decryption_tuple)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/cloud_api_delegator.py", line 366, in GetObjectMedia
    decryption_tuple=decryption_tuple)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/gcs_json_api.py", line 1225, in GetObjectMedia
    decryption_tuple=decryption_tuple)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/gcs_json_api.py", line 1358, in _PerformDownload
    use_chunks=False)
  File "/snap/google-cloud-sdk/179/platform/gsutil/third_party/apitools/apitools/base/py/transfer.py", line 486, in GetRange
    response = self.__ProcessResponse(response)
  File "/snap/google-cloud-sdk/179/platform/gsutil/third_party/apitools/apitools/base/py/transfer.py", line 433, in __ProcessResponse
    self.__progress += response.length
  File "/snap/google-cloud-sdk/179/platform/gsutil/third_party/apitools/apitools/base/py/http_wrapper.py", line 195, in length
    return int(self.info.get('content-length'))
ValueError: invalid literal for int() with base 10: '104857600, 94911513'

invalid literal for int() with base 10: '104857600, 94911513'
Exception in thread Thread-14:
Traceback (most recent call last):
  File "/snap/google-cloud-sdk/179/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/snap/google-cloud-sdk/179/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/daisy_chain_wrapper.py", line 200, in PerformDownload
    decryption_tuple=self.decryption_tuple)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/cloud_api_delegator.py", line 366, in GetObjectMedia
    decryption_tuple=decryption_tuple)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/gcs_json_api.py", line 1225, in GetObjectMedia
    decryption_tuple=decryption_tuple)
  File "/snap/google-cloud-sdk/179/platform/gsutil/gslib/gcs_json_api.py", line 1358, in _PerformDownload
    use_chunks=False)
  File "/snap/google-cloud-sdk/179/platform/gsutil/third_party/apitools/apitools/base/py/transfer.py", line 486, in GetRange
    response = self.__ProcessResponse(response)
  File "/snap/google-cloud-sdk/179/platform/gsutil/third_party/apitools/apitools/base/py/transfer.py", line 433, in __ProcessResponse
    self.__progress += response.length
  File "/snap/google-cloud-sdk/179/platform/gsutil/third_party/apitools/apitools/base/py/http_wrapper.py", line 195, in length
    return int(self.info.get('content-length'))
ValueError: invalid literal for int() with base 10: '104857600, 95509541'

invalid literal for int() with base 10: '104857600, 95509541'`

I tried to run rsync multiple times and the same issue happened.

Any ideas on how to solve the problem guys?

Cheer,
Fabio

**Comment by rrauber** [comment]
Hi Fabio!

This looks a lot like #1110, which I have a PR out to fix (#1231). The error occurs when a download is interrupted partway through due to a flaw in gsutil's retry logic. Failures of this sort should be intermittent though, so since you can reproduce it consistently there might be something else wrong. Have you seen any successful runs? If not, does the issue always occur for the same file?

**Comment by fabio-rigato** [comment]
Dear @rrauber,
thanks for your reply.
I can confirm that the issue I am facing is with the same files.
In particular, I'm sync between GCS and S3.

Please let me know if you have any suggestions on how to sync that files :)

Cheers,
Fabio
